staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -95,6 +112,10 @@ type FeatureGate interface {
 	// set on the copy without mutating the original. This is useful for validating
 	// config against potential feature gate changes before committing those changes.
 	DeepCopy() MutableFeatureGate
+	// EmulationVersion returns the version the feature gate is set to emulate.
+	// If set, the feature gate would enable/disable features based on
+	// feature availability and pre-release at the emulated version instead of the binary version.
+	EmulationVersion() *version.Version
```

Comment:
I just noticed that we have the getter for emulation version on feature gate, but don't have the setter on MutableFeatureGate. Instead the setter is on MutableVersionedFeatureGate` 

 This begs the question of if the getter should be on FeatureGate? I think I like having the setter on MutableVersionedFeatureGate, but there is  no corresponding `VersionedFeatureGate` interface for the getter to live in, should we introduce one? Does it make a large mess to do so?

Reply:
done.